contributors/design-proposals/pod-preemption.md

```diff
@@ -0,0 +1,187 @@
+   -  The scheduler may still preempt Best Effort pods for reasons other than releasing their resources. For example, it may preempt a Best Effort pod in order to satisfy affinity rules of the pending pod.
+   -  The amount that needs to be freed (when the issue is resources) is request of the pending pod. 
+
+## Preemption - Eviction workflow
+
+"Eviction" is the act of killing one or more pods on a node when the node is under resource pressure. Kubelet performs eviction. The eviction process is described in separate document by sig-node, but since it is closely related to the "preemption", we explain it briefly here.  
+Similar to preemption, the lowest priority pods are chosen for eviction first. The difference between preemption and eviction is that when pods with the same priority are considered for eviction, the one with the highest percentage of usage over "requests" is the one that is evicted first. In other words, Kubelet uses priority for eviction and breaks ties by usage over requests.  
+This implies that Best Effort pods are more likely to be evicted among a set of pods with the same priority. The reason is that any amount of resource usage by Best Effort pods translates into a very large percentage of usage over "requests", as Best Effort pods have zero requests for resources. So, while scheduler does not preempt Best Effort pods for releasing resources on a node, it is likely that these pods are evicted by the Kubelet after scheduler schedules a higher priority pod on the node.  
+Here is an example:
+
+1. Assume we have a node with 2GB of usable memory by pods.
+1. The node is running a burstable pod that uses 1GB of memory and 500MB memory request. It also runs a best effort pod that uses 1GB of memory (and 0 memory request). Both of the pods have priority of 100.
+1. A new pod with priority 200 arrives and asks for 2GB of memory.
+1. Scheduler knows that it has to preempt the burstable pod. From scheduler's point of view, the best effort pod needs no resources and its preemption will not release any resources.
+1. Scheduler preempts the burstable pod and schedules the high priority pending pod on the node.
+1. The high priority pod uses more than 1GB of memory. Kubelet detects the resource pressure and kills the best effort pod.
+
+So, best effort pods may be killed to make room for higher priority pods, although the scheduler does not preempt them directly.  
+Now, assume everything in the above example, but the best effort pod has priority 300. In this scenario, scheduler schedules the pending pod with priority 200 on the node, but it is evicted by the Kubelet, because the best effort pod has a higher priority. Given this scenario, scheduler should avoid the node and should try scheduling the pod on a different node if the pod is evicted by the Kubelet.
+
```

Comment:
I think this is more of a nice-to-have than absolutely necessary. But actually we do have "avoid previous node" already, see #20699, but note that (1) it's not used anywhere yet so is definitely an alpha feature, and (2) it's on the set granularity rather than pod granularity for the reason you mentioned (pods do not get re-created, so it's nontrivial to determine that a particular pod you are trying to schedule is the replacement pod for some particular other pod that terminated).